### PR TITLE
Soback 30 refactor of search function

### PR DESCRIPTION
**Pull request by jesperlauridsen**
So, here's whats done:

preNormalizedQuery moved into the store
SearchUtil.js mixin created
All search related functionality moved into SearchUtil.js
Searchfunction (that handles which kind of search should be triggered split into components
Search-function put into use the places that were triggering variants of the same code before

### CODE REVIEW

File: src/js/src/views/SolrWayback.vue [hunk 1]
```diff
@@ -67,29 +69,13 @@ export default {
...
-        queryString.substring(queryString.length-1, queryString.length) === '"' ? queryString.slice(0,-1) : null
-        this.updateQuery(queryString)
-      }
-      this.requestUrlSearch({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
-      this.requestFacets({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
-    } 
-    else {
-      this.requestSearch({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
-      this.requestFacets({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
-    }
```
Comment (jorntx) [inline src/js/src/views/SolrWayback.vue:92]: So good with the refactoring - this refactoring will certainly come in handy when fixing the routing issues.

**Review by jorntx — COMMENTED** [review]
The refactoring is approved! 

But I'm unsure if it has introduced a small bug in the search for image function (in the image search result).

Try to perform an image search and select a random result and pres the 'SEARCH FOR IMAGE" link. Nothing seems to happen (other than the query is updated). 

Can you verify this and get back to me?

File: src/js/src/components/SearchBox.vue [hunk 2]
```diff
@@ -97,43 +98,10 @@ export default {
...
-        }
-        else {
-          this.requestSearch({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.requestFacets({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
-        }
+      this.$router.history.current.query.grouping === 'true' ? this.updateFutureSolrSettingGrouping(true) : this.updateFutureSolrSettingGrouping(false)
+      this.$router.history.current.query.imgSearch === 'true' ? this.updateFutureSolrSettingImgSearch(true) : this.updateFutureSolrSettingImgSearch(false)
+      this.$router.history.current.query.urlSearch === 'true' ? this.updateFutureSolrSettingUrlSearch(true) : this.updateFutureSolrSettingUrlSearch(false)
+      this.determineNewSearch(this.futureQuery)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:104]: Check - fixed now.

File: src/js/src/mixins/SearchUtils.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        else if(this.solrSettings.urlSearch) {
+          this.deliverUrlSearchRequest(futureQuery)
+        }
+        else {
+          this.deliverSearchRequest(futureQuery)
+        }
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/mixins/SearchUtils.js]: :+1:

File: src/js/src/mixins/SearchUtils.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            queryString.substring(queryString.length-1, queryString.length) === '"' ? queryString = queryString.slice(0,-1) : null
+          }
+          else {
+            queryString = futureQuery
+          }
+          return queryString
+    },
+    // Method to fire off a search (and deciding which kind it is)
+    determineNewSearch(futureQuery) {
+      if(this.checkForChangesInSolrSettings() || this.checkForChangesInQuery(futureQuery)) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/mixins/SearchUtils.js]: :+1:

File: src/js/src/mixins/SearchUtils.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      clearResults:'clearResults',
+      requestSearch:'requestSearch',
+      requestImageSearch:'requestImageSearch',
+      requestUrlSearch:'requestUrlSearch',
+
+    }),
+    ...mapActions('Notifier', {
+      setNotification: 'setNotification'
+    }),
+    validateUrl(testString) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/mixins/SearchUtils.js]: :+1:

File: src/js/src/components/SearchBox.vue [hunk 6]
```diff
@@ -97,43 +98,10 @@ export default {
...
-            type: 'error',
-            timeout: false
-          })
-          }
-        }
-        else {
-          this.requestSearch({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.requestFacets({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
-        }
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:136]: :+1:

File: src/js/src/components/SearchBox.vue [hunk 7]
```diff
@@ -154,60 +122,13 @@ export default {
...
-      }
-    },
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:210]: :+1:

File: src/js/src/views/SolrWayback.vue [hunk 8]
```diff
@@ -67,29 +69,13 @@ export default {
...
+    this.determineNewSearch(to.query.q)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/views/SolrWayback.vue]: Yea, fixed now.

File: src/js/src/views/SolrWayback.vue (hunk 1 above)
Comment (jesperlauridsen) [reply src/js/src/views/SolrWayback.vue:92]: :+1:

**Comment by jesperlauridsen** [comment]
I found the bug - query was being updated too fast, so when i checked if anything had changed, it returned to. Fixed now, caused problems for both image related searches.

**Comment by jorntx** [comment]
Looks good now and tests out fine - needs some merge conflict resolving 🦖

**Comment by jorntx** [comment]
Just found another issue - after using the SEARCH FOR IMAGE link reload does not work anymore.

**Comment by jorntx** [comment]
Hhmm - reload seems to be broken no matter what search you have performed.

**Comment by jorntx** [comment]
Branch is now mergeable - would you @jesperlauridsen prefer this merged into newfrontend now or do you want to wait?

**Comment by jesperlauridsen** [comment]
Please wait for an update tomorrow morning!

**Comment by jorntx** [comment]
Roger!

**Comment by jesperlauridsen** [comment]
Not relevant with the new branch 'Soback 59 and 30 combined' having the changes instead.

**Pull request closed without merging**